interface/backend/cases/utils.py

```diff
@@ -0,0 +1,14 @@
+import inspect
+from enum import Enum
+
+
+class ChoiceEnum(Enum):
```

Comment:
There are a number of Django enum fields - did you look into them at all? They seem to do this stuff a little nicer, or at least with less bus-factor!